File: lesson_1/code/task_2_2_malthusian_explicit.m
% Population parameters
    P0 = 2.50; % Intial popoulation 
    r = 0.027; % Growt rate 2.7%

% Timescale
    t_start = 0;  t_end = 70;  t_diff = 1; 

% Years of interest
    t_expl= t_start:t_diff:t_end;

 % Results
    P_expl= P0*exp(r*t_expl);

% Plot
    plot(t_expl, P_expl, LineWidth=2)
    xlabel("Time (years)");  
    ylabel("Population (millions)"), 
    ylim([0, 20])

File: lesson_1/code/task_2_2_malthusian_explicit.py
import numpy as np
import matplotlib.pyplot as plt

P0 = 2.50
r = 0.027
t_start, t_end, t_diff = 0, 70, 1

t_expl = np.arange(t_start, t_end + t_diff, t_diff)
P_expl = P0 * np.exp(r * t_expl)

plt.plot(t_expl, P_expl, linewidth=2)
plt.xlabel("Time (years)")
plt.ylabel("Population (millions)")
plt.ylim(0, 20)
plt.show()